File: tutorial-1.19/src/scala/scala/org/quiltmc/wiki/TutorialMod.scala
package org.quiltmc.wiki

import net.minecraft.item.Item
import net.minecraft.util.Identifier
import net.minecraft.registry.{Registries, Registry}
import org.quiltmc.loader.api.ModContainer
import org.quiltmc.qsl.base.api.entrypoint.ModInitializer
import org.quiltmc.qsl.item.setting.api.QuiltItemSettings

// Can be made an object with a language adapter for scala
class TutorialMod extends ModInitializer:
  override def onInitialize(mod: ModContainer): Unit =
    //@start register_item
    Registry.register(Registries.ITEM, Identifier(mod.metadata.id, "example_item"), TutorialMod.EXAMPLE_ITEM)
    //@end register_item

object TutorialMod:
  //@start declare_item
  final val EXAMPLE_ITEM: Item = Item(QuiltItemSettings())
  //@end declare_item

File: tutorial-1.19/src/java/java/org/quiltmc/wiki/TutorialMod.java
package org.quiltmc.wiki;

import net.minecraft.item.Item;
import net.minecraft.registry.Registries;
import net.minecraft.util.Identifier;
import net.minecraft.registry.Registry;
import org.quiltmc.loader.api.ModContainer;
import org.quiltmc.qsl.base.api.entrypoint.ModInitializer;
import org.quiltmc.qsl.item.setting.api.QuiltItemSettings;

public class TutorialMod implements ModInitializer {
	//@start declare_item
	public static final Item EXAMPLE_ITEM = new Item(new QuiltItemSettings());
	//@end declare_item

	@Override
	public void onInitialize(ModContainer mod) {
		//@start register_item
		Registry.register(Registries.ITEM, new Identifier(mod.metadata().id(), "example_item"), EXAMPLE_ITEM);
		//@end register_item
	}
}
